crates/core/src/host/wasm_common/module_host_actor.rs

```diff
@@ -275,12 +275,13 @@ impl<T: WasmModule> Module for WasmModuleHostActor<T> {
         let db = &*self.worker_database_instance.relational_db;
         db.with_auto_commit(|tx| {
             let tables = db.get_all_tables(tx)?;
-            for table in tables {
-                if table.table_name != table_name {
-                    continue;
-                }
-
-                db.clear_table(tx, table.table_id)?;
+            let clear_ids = tables
+                .iter()
+                .filter(|t| t.table_name == table_name)
+                .map(|t| t.table_id)
+                .collect::<Vec<_>>();
```

Comment:
Is there a stateful reason to collect these? ie, if you called `clear_table` on the first one, it would in some way invalidate things later on?